### PR TITLE
CAR pixelization

### PR DESCRIPTION
**Pull request by damonge**
My own version of `rectpix_both`, to merge before finishing up and merging that one on master

### CODE REVIEW

**Comment by xzackli** [comment]
This looks great! I'll look at it in more detail this evening (about to meet with Jo), and I'll also take a look nmt_mask's apodization routines.

**Comment by damonge** [comment]
Great! I'd say don't worry about the apodization routines too much. I actually want to revamp those, even the HEALPix ones, so that could be done as a separate PR once the CAR stuff is dealt with.

File: src/healpix_extra.c [hunk 1]
```diff
@@ -136,7 +120,116 @@ flouble *he_read_healpix_map(char *fname,long *nside,int nfield)
...
+  fits_read_key(fptr, TDOUBLE, "CDELT1", &Delta_phi, comment, &status);
+  fits_read_key(fptr, TDOUBLE, "CDELT2", &Delta_theta, comment, &status);
+  fits_read_key(fptr, TDOUBLE, "CRVAL1", &phi0, comment, &status);
+  fits_read_key(fptr, TDOUBLE, "CRVAL2", &theta0, comment, &status);
+  free(comment);
+  
+  flipx = Delta_phi < 0;
+  flipy = Delta_theta > 0;
+
+  full_x = (int) round(360.0 / Delta_phi); // have to fill in ring
```
[the remark this answers is not in the document]
Comment (xzackli) [reply src/healpix_extra.c]: You're right, we should probably store only the absolute value of Delta_phi and only consider the sign when loading the file into memory.

File: src/healpix_extra.c [hunk 2]
```diff
@@ -136,7 +120,116 @@ flouble *he_read_healpix_map(char *fname,long *nside,int nfield)
...
+
+  sky_info->nx = full_x;
+  sky_info->ny = axes[1];
+  sky_info->npix = full_x * axes[1];
+  sky_info->Delta_phi = Delta_phi * M_PI / 180.0;
+  sky_info->Delta_theta = Delta_theta * M_PI / 180.0;
```
[the remark this answers is not in the document]
Comment (xzackli) [reply src/healpix_extra.c]: Yes! Definitely clearer to store the absolute value only.

File: src/healpix_extra.c [hunk 3]
```diff
@@ -907,13 +1168,24 @@ void he_alter_alm(int lmax,double fwhm_amin,fcomplex *alm_in,fcomplex *alm_out,
     free(beam);
 }
 
-void he_map_product(int nside,flouble *mp1,flouble *mp2,flouble *mp_out)
+flouble he_get_pix_area(nmt_curvedsky_info *cs,long i)
```
[the remark this answers is not in the document]
Comment (xzackli) [reply src/healpix_extra.c]: I agree! The addition of several `fabs` in the file loading is good, as you suggested.

File: src/healpix_extra.c [hunk 4]
```diff
@@ -907,13 +1168,24 @@ void he_alter_alm(int lmax,double fwhm_amin,fcomplex *alm_in,fcomplex *alm_out,
...
+{
+  if(cs->is_healpix) {
+    return M_PI/(3*cs->n_eq*cs->n_eq);
+  }
+  else {
+    flouble theta1 = i * cs->Delta_theta + cs->theta0;
```
[the remark this answers is not in the document]
Comment (xzackli) [reply src/healpix_extra.c]: I think that you need the spherical coordinate theta of a pixel, in order to calculate the area of a particular pixel? The CAR maps are not full sky, I only fill in the phi direction. The first ring is located at theta0, so you need to include it in the offset.

`i` is the ring index here?

File: src/healpix_extra.c [hunk 5]
```diff
@@ -685,7 +902,21 @@ static void sht_wrapper(int spin,int lmax,int nside,int ntrans,flouble **maps,fc
...
+  if(cs->is_healpix)
+    sharp_make_weighted_healpix_geom_info(cs->n_eq,1,NULL,&geom_info);
+  else {
+    // first nrings is total
+    flouble Delta_theta = fabs(cs->Delta_theta);
+    int tot_sphere_rings = round(M_PI / Delta_theta) + 1;
+    flouble theta1 = cs->theta0 - cs->ny * Delta_theta;
+    int first_ring = round(theta1 / Delta_theta) + 1;
+    sharp_make_cc_geom_info_stripe(tot_sphere_rings,
+				   cs->nx, cs->phi0, // DEGREES OR RADIANS??
```
Comment (xzackli) [inline src/healpix_extra.c]: Probably should remove this comment at some point, libsharp operates **in radians**.

Comment (damonge) [reply src/healpix_extra.c]: will remove in my next push

**Comment by damonge** [comment]
@xzackli this is ready for a quick review when you have time

**Pull request merged**